chapters/kodi/index.md

```diff
@@ -0,0 +1,269 @@
+
+## 4. Development View - ***The Building Blocks of Kodi***
+The Kodi foundation aims to work with independent software modules, as stated on the [Kodi Wiki website](http://kodi.wiki/view/XBMC_development):
+> *"Kodi should still compile and run if a non-essential module/library is disabled or removed."*
+
+This is one of Kodi's general guidelines that actually fits in the [Rozansky and Woods](http://www.viewpoints-and-perspectives.info) requirements for successful architecture of complex software. Introducing modules improves the scalability and maintainability of the source code, especially when the software is being used and compiled while the developing and maintaining continues. In the case that one of the modules has failures the software can still be used. They serve as the 'building blocks' of the total architecture that are interacting with each other and distinguishing them from each other based on the functional activities performed. Subsequently these modules are classified in functional groups which are independent from each other. 
+
+Starting with discussing the [modular design](#modular-design) that form the 'building blocks' of Kodi and illustrate these with an overview to gain a deeper understanding of the architecture of Kodi. Furthermore, the dependencies of and interactions of the modules and layers are discussed. 
+
+### 4.1. Modular Design
+
+<p align="center">
+   <img src="images/KodiDevelopmentView.png" width=700px>
+   <p align="center"><i>Figure 5: Kodi's 'building blocks'</i></p>
+</p>
+
+Firstly in Figure 5, the **yellow** building block encloses all files and packages about the [skins](http://kodi.wiki/view/Category:Skins), translations, fonts, and GUI of the Kodi Media Player. This is one of the most popular modules for the external developers, while it also provides a perfect example for the visualisation of the working of modules within a greater software architecture.  
+
+Secondly, the **purple** Interface building block includes all the files and packages concerning among others the add-ons, streaming clients, event servers and external libraries. In this case, it is again possible to visualise how the rest of the system continues to work while an add-on is not working properly. The add-ons are mostly designed by external developers, and mistakes are more likely to occur since they have less knowledge of the system. 
+
+An interesting and very popular add-on which was also a [DelftStudentsOnSoftwareArchitecture team in 2014](http://delftswa2014.github.io/chapters/sick-beard/index.html), is [Sickbeard](https://seo-michael.co.uk/how-to-install-vdubt25-add-on-kodi-xbmc/). This add-on enables to download a shows episodes once they are published on torrent trackers. 
```

Comment:
@avandeursen not sure if we want to link here? Last year we didn't really make a public announcement